contracts/voting-verifier/src/contract.rs

```diff
@@ -23,6 +23,7 @@ pub fn instantiate(
         voting_threshold: msg.voting_threshold,
         block_expiry: msg.block_expiry,
         confirmation_height: msg.confirmation_height,
+        source_chain: msg.source_chain,
```

Comment:
It's already linked to a source chain. There is a source_gateway_address field already, and the contract is meant to work for a single source chain.

Reply:
That's not necessarily true cuz we can use CREATE2 to deterministically deploy gateways onto EVM chains so that they all have the exact same address. However, if we decide to link this contract to a specific chain, then the source chain check at https://github.com/axelarnetwork/axelar-amplifier/blob/b5b50fb142795700288a7c7df9f19aae02d8276c/contracts/voting-verifier/src/contract.rs#L56 absolutely needs to change. At https://github.com/axelarnetwork/axelar-amplifier/blob/b5b50fb142795700288a7c7df9f19aae02d8276c/contracts/voting-verifier/src/execute.rs#L128, it also has to change so that source chain isn't derived from messages.

Reply:
The gateways today though don't all have the same address. We can deploy with the same address, but I wouldn't want to add that as a requirement at the protocol layer. 

We also have just generally been planning to have one voting verifier per chain, for now at least. We can revisit that in the future, because there are benefits to having just one voting verifier. But for now, adding the source chain here doesn't break anything about the model we've been working off of.

Reply:
@cjcobb23 I am OK with either way, but I need to see consistency within the code of this particular contract. Right now, the messages verification part isn't consistent with the fact that this contract is linked to one single source chain.